qiskit/converters/circuit_to_gate.py

```diff
@@ -92,14 +81,13 @@ def find_bit_position(bit):
     if gate.num_qubits > 0:
         q = QuantumRegister(gate.num_qubits, 'q')
 
+    qubit_map = {bit: q[idx] for idx, bit in enumerate(circuit.qubits)}
```

Comment:
I'm wondering if it makes sense to carry this around in the circuit object itself (maybe as a private). Pretty much anything that is going from a circuit to something else is now building this dict.